### PR TITLE
Use org_id instead of account for queries and Identity

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [ESSNTL-2529](https://issues.redhat.com/browse/ESSNTL-2529), [ESSNTL-3112](https://issues.redhat.com/browse/ESSNTL-3112), and [ESSNTL-3147](https://issues.redhat.com/browse/ESSNTL-3147).
Switches to using `org_id` instead of account as the primary required field for both the Identity and DB queries. It also removes the tenant translator, since it will no longer be used, as `org_id` is now required. It also makes `account`/`account_number` optional, and deprecates the field in the API spec.

Finally, it updates the pendo-syncher to report hosts by org_id instead of account, since the hosts we're sent will eventually not have an account. This may make the data strange in Pendo for a while, but if someone's actually still watching that data in Pendo, they can let us know and we'll address potential solutions at that time.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [x] Tests: validate exceptions and failure scenarios
- [x] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by thearifismail — DISMISSED** [review]
LGTM

File: swagger/api.spec.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ paths:
         - hosts
       summary: Read the entire list of hosts
       description: >-
-        Read the entire list of all hosts available to the account.
```
Comment (fstavela) [inline swagger/api.spec.yaml:15]: I would maybe keep the word "account" in the endpoints documentation. They will still return hosts from the "account", it's just that these accounts will no longer be identified by account_number, but by org_id. @kruai what do you think?

Comment (kruai) [reply swagger/api.spec.yaml]: OK @fstavela, I've changed them back to using "account" and made a few fixes for clarity.

**Review by fstavela — DISMISSED** [review]
Works fine together with https://github.com/RedHatInsights/xjoin-search/pull/159 , but I think we should wait with merging until all reporters start sending org_id in both identity and payload.

**Review by thearifismail — DISMISSED** [review]
LGTM

**Comment by kruai** [comment]
I have confirmed that both yupana and rhsm-conduit - the two reporters we were waiting on - have merged their `org_id` fixes to Stage, and will have them in Prod by Monday. I'm going to go ahead and merge this so we can unblock things like [this](https://github.com/RedHatInsights/insights-puptoo/pull/223).

**Pull request merged**